tests/api_resources/abstract/test_api_resource.py

```diff
@@ -101,3 +113,118 @@ def test_raise_on_incorrect_id_type(self):
         for obj in [None, 1, 3.14, dict(), list(), set(), tuple(), object()]:
             with pytest.raises(stripe.error.InvalidRequestError):
                 self.MyResource.retrieve(obj)
+
+    def test_class_method_does_not_forward_options(
+        self, http_client_mock
+    ):
+        http_client_mock.stub_request(
+            "get",
+            "/v1/myresources/foo",
+            rbody='{"id": "foo"}',
+        )
+
+        key = "newkey"
+        stripe_version = "2023-01-01"
+        stripe_account = "acct_foo"
+
+        resource = self.MyResource.construct_from(
+            {"id": "foo"},
+            key=key,
+            stripe_version=stripe_version,
+            stripe_account=stripe_account,
+        )
+
+        resource.retrieve("foo")
+
+        http_client_mock.assert_requested(
+            "get",
+            path="/v1/myresources/foo",
+            api_key=stripe.api_key,
+            stripe_version=stripe.api_version,
+            extra_headers={"Stripe-Account": None},
+        )
+
+    def test_class_method_prefers_method_arguments(
+        self, http_client_mock
+    ):
+        http_client_mock.stub_request(
+            "get",
+            "/v1/myresources/foo",
+            rbody='{"id": "foo"}',
+        )
+
+        resource = self.MyResource.construct_from(
+            {"id": "foo"},
+            key="oldkey",
+            stripe_version="2000-01-01",
+            stripe_account="acct_foo",
+        )
+
+        resource.retrieve("foo", api_key="newkey", stripe_version="2023-01-01", stripe_account="acct_bar")
+
+        http_client_mock.assert_requested(
+            "get",
+            path="/v1/myresources/foo",
+            api_key="newkey",
+            stripe_version="2023-01-01",
+            stripe_account="acct_bar",
+        )
+
+    def test_instance_method_forwards_options(
+        self, http_client_mock
+    ):
+        http_client_mock.stub_request(
+            "get",
+            "/v1/myresources/foo",
+            rbody='{"id": "foo"}',
+        )
+
+        key = "newkey"
+        stripe_version = "2023-01-01"
+        stripe_account = "acct_foo"
+
+        resource = self.MyResource.construct_from(
+            {"id": "foo"},
+            key=key,
+            stripe_version=stripe_version,
+            stripe_account=stripe_account,
+        )
+
+        resource.refresh()
+
+        http_client_mock.assert_requested(
+            "get",
+            path="/v1/myresources/foo",
+            api_key=key,
+            stripe_version=stripe_version,
+            stripe_account=stripe_account,
+        )
+
+    def test_instance_method_prefers_method_arguments(
```

Comment:
What a bout an instance call on a resource returned by a method call with extra parameters.
```python
acc = Account.retrieve(, stripe_account)
acc.delete()
```

Reply:
I guess that's covered by refresh example above.

Reply:
Good point, I went and added a few more assertions to test_retrieve_and_refresh to make it more complete.